### ISSUE TITLE
Carriers rarely get a chance to repair their fighters while in combat

### ISSUE THREAD

**Issue by Amazinite** [endless-sky/endless-sky#9429]
### Problem Description

The current behavior for carriers is that if they have excess energy, shield generation, or hull repair, they'll give that excess to their docked fighters. For many fighters, this is their only means of shield generation and hull repair. The problem is that the carriers themselves will engage in combat, causing them to become damaged and use up their energy. This means that in practice, carriers never have excess to provide to their fighters, meaning that fighters that managed to return to their carrier after being damaged never get redeployed back into combat.

### Related Issue Links

#4034

### Desired Solution

Have a setting where carriers save a fraction of their energy and repairs to give to their fighters. As an example, say a carrier is lightly damaged, it could use only 50% of its repairs for itself and provide the remaining 50% to any docked fighters that need it, instead of only ever providing excess. Although if the carrier is heavily damaged, perhaps it keeps all 100% for itself.

### Alternative Approaches

Another approach that I believe @Quantumshark has mentioned is to allow fighters to share resources with one another. Fighters still provide generation to themselves while docked, but do nothing with their excess. By allowing fighters to share resources with one another, they could put their excess to use helping the other docked fighters get up and running. This approach wouldn't help for fighters that don't have repairs anyway, but it would still be a nice bonus in general for fighters.

We could also have carriers attempt to stay out of combat, therefore meaning they won't be as damaged and energy sapped and will be more likely to have excess for their fighters.

**Comment by Zitchas** [comment]
The very core of this issue is that we don't have carriers, we have CQC battleships that just so happen to carry a few fighters to support them. We really need to rebuild carriers to *not* engage in CQC and actually have the majority of their offensive power be in their fighters, **not** their own personal onboard weaponry. Until we do that, carriers aren't a thing we have in this game.

As an alternate approach, in Vitalchip's build there are defensive hardpoints which a) aren't manually controllable, b) are automatically opportunisitic and unfocused, and c) aren't considered by the AI for the purposes of determining engagement range. This means that these defensive weapons are highly useful as anti-fighter weaponry, but very poorly effective as offensive weaponry. This would considerably help the AI keep its distance, and therefore stay alive longer, take less damage, and thus as a result it would actually do its job repairing fighters. This would mean that the carrier+fighter team would actually work as we expect a carrier+fighter wing is expected to work, and less like a battleship with a squad of ultra-fragile disposable distractions to accompany it.

Carriers need:
- Less personal offensive capacity
- Less speed to get into melee combat with
- Artillery *support* weapons (not stand-alone awesome artillery, but rather *support* artillery that is only useful on targets pinned down by fighters)
- More fighters. Enough to actually get a cycle of "fighters fighting, fighters returning, fighters being repaired, and fighters travelling to the target." As far as this goes, being able to launch fighters individually or in small batches conveniently would be useful so they can hit in waves.
- More space devoted to fighter-support facilities (reloading, energy supply, repair, etc.) and less of these being on the fighters.

**Comment by EjoThims** [comment]
Splitting shared regen sources across all applicable targets based on target % would be a great alternative to the current behavior of only sharing excess, but this is indeed ultimately a problem of the biggest, most specialized carriers still being hybrids with their own offensive capabilities instead of relying primarily on their fighters.

In addition to the various suggestions above from @Zitchas , one small, immediately actionable change we could make in this regard is the long considered doubling of capacity for the Nest line, coupled with switching their load outs entirely for AM, regen, and ammo storage (pirates still fielding more directly offensive variants/loadouts would make sense though).

This wouldn't require remodeling or new sprite work, and would be a boon to FW fleet composition, which has generally suffered since the changes to human weaponry, helping to emphasize both the differences in battle tactics and available resources from the lore side of things.

More importantly, it would give us at least a single line of dedicated specialist carriers (though still with relatively limited capacity even after doubling it) as a testbed for AI usage.

**Comment by Quantumshark** [comment]
You can build a carrier to use the long-range missile boat AI, and that will have it take a long-range position from which it can safely deploy fighters, at least as long as it's not being chased down by something faster than it. That's not going to make a huge difference for carriers that aren't very fast, though - the Nest/Roost/Skein with their painfully low top speeds aren't outrunning any pursuers any time soon.
Really, if you want carriers to be maintaining a distance from their opponents, they need _more_ speed, not less.

In terms of the proper carrier vs battleship with a handful of fighters question, there are alien ships, both in-game and in the works, that fit the proper carrier role quite well - the TF109, and Azure's Atenuatia, to give two examples. So the issue isn't really that we don't have proper carriers, it's that we don't have proper carriers _in human space_. A doubling of the fighter capacity of the Nest/Roost/Skein as you suggest would solve that issue, providing an eight-bay carrier available to mid-range players in the form of the Roost, and also making the Skein a more attractive reward for joining the FW.

**Comment by Hecter94** [comment]
We've been talking a lot about how we want to tweak fighters to improve their utility, so what if we just allowed a Carrier to repair their docked fighters without diminishing its repair?
Rather than require the Carrier to use its own Shield/Hull regen and slow down its repair rate, what if it was simply able to use any excess hull/shield regen plus, say, 10% of what it is currently using?

So, for example: 
A carrier with 100 shield generation that it isn't currently using would be able to give 100 shield generation to its fighters. 
A carrier with 100 shield generation that it *is* currently using would be able to give 100 shield generation to itself and ten shield generation to its fighters. 
A carrier with 100 shield generation that it isn't using and 50 hull repair that it *is* using would give 100 shield generation and five hull repair to its fighters. 

This effectively means that a Carrier will *always* be able to repair its docked fighters, just slowly if it's currently using all of its generation for itself, and we could tweak the amount applied if 10% seems *too* slow. 
It removes the need for a preference, doesn't nerf the repair rate of a Carrier that's currently under attack, and gives a slight overall buff to the utility of Carriers.

**Comment by Zitchas** [comment]
I don't think that letting carriers "cheat" and double up on regen would be very useful. I'd much rather see fighter-specific repair facilities implemented as outfits that *only* repair fighters, regardless of what regen the carrier may or may not want for itself. I'd say that such fighter-specific repair facilities, being equipment in bays that have easier access to the whole fighter, would be significantly faster at doing repairs than regular self-regen outfits. And takes up space and adds mass to carriers, which they desperately need. (as in, they need more mass and more space taken up with fighter-related stuff)

**Comment by mOctave** [comment]
Honestly, even just having a fighter hull repair outfit and corresponding attribute would make sense. You could set the repair rate to be quite fast, and then carriers would have to choose what to be good at.

Of course, the other side of this that hasn't been being discussed too much (probably since it's very similar) is drones.

**Comment by all-of-olive** [comment]
I would like to echo the suggestion for fighter and drone repair/regen to be transitioned to a series of outfits! It helps flesh out the role of dedicated carrier hulls, while also forcing outfitting tradeoffs for hybrid vessels like Cruisers, and even creates some new niches and playstyles (especially in the early game) that would be less balanced under the current system.

Imagine a single-fighter hyperdrive sled: slow, fragile, unarmed, zero maintenance capability—basically only useful for moving fighters from point A to point B. Vulnerable in transit, and with a very good chance of getting taken out in the fray if you try to get cute and use them offensively—forcing you to either write off the cost of both sled and fighter, or buy *another* sled for retrieval. 

The next step up would be ultralight carriers: combat hulls that can accommodate fighters and drones via docking clamps and external racks, but that lack the outfitting space to maintain them. Think along the lines of a Free Worlds combat refit of the Sunder.

Early-game hybrid vessels become a more viable option, too. A pricier Star Barge variant that packs a single drone for emergencies; an even more ponderous Freighter that can support a Barb for defense—or a Boxwing for ore collection during mining ops.
